src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -580,7 +580,13 @@ static void* VirtualReserveInner(size_t size, size_t alignment, uint32_t flags,
     }
 
     size_t alignedSize = size + (alignment - OS_PAGE_SIZE);
-    void * pRetVal = mmap(nullptr, alignedSize, PROT_NONE, MAP_ANON | MAP_PRIVATE | hugePagesFlag, -1, 0);
+    int mapflags = MAP_ANON | MAP_PRIVATE | hugePagesFlag;
+#if defined(MAP_NORESERVE) && defined(__sun)
```

Comment:
nit: to be consistent with TARGET_LINUX and TARGET_APPLE in nearby context.
```suggestion
#if defined(MAP_NORESERVE) && defined(TARGET_SUNOS)
```

Reply:
OK, though I see a lot of cases where the code currently uses `__sun` as the pre-processor symbol to check.
If we're going to use TARGET_SUNOS instead, we should probably change them all.

Personally, I'd skip it for now, and maybe later add a TARGET_ILLUMOS (less confusing name)
and then go change all the __sun checks.

Reply:
np, it's ok to keep it as-is.

* we apply "what's used in the nearby context" logic for consistency. So when there is  {HOST,TARGET}_XX used in a file, those are preferred.
* TARGET_SUNOS (more permissive) is better than TARGET_ILLUMOS unless we can determine solaris implementation is going to differ.

Reply:
Let me try the suggest on my setup to make sure it works OK.

Reply:
Yup, that works fine.  Push it! :)